src/ledger/LedgerManagerImpl.cpp

```diff
@@ -618,6 +604,48 @@ LedgerManagerImpl::hasLastClosedSorobanNetworkConfig() const
     return mLastClosedLedgerState->hasSorobanConfig();
 }
 
+std::chrono::milliseconds
+LedgerManagerImpl::getExpectedLedgerCloseTime() const
+{
+    releaseAssert(threadIsMain());
+
+#ifdef BUILD_TESTS
+    auto const& cfg = mApp.getConfig();
+    // Always check for testing override first
+    if (cfg.ARTIFICIALLY_SET_CLOSE_TIME_FOR_TESTING)
+    {
+        return std::chrono::milliseconds{
+            cfg.ARTIFICIALLY_SET_CLOSE_TIME_FOR_TESTING * 1000};
+    }
+    if (cfg.ARTIFICIALLY_ACCELERATE_TIME_FOR_TESTING)
+    {
+        return std::chrono::milliseconds{1000};
+    }
+#endif
+
+    // If a service calls this function during startup, it's possible that we
+    // have not yet set last closed ledger state, so return the "default" value.
+    // This is fine, since any protocol or consensus related functions that have
+    // a strong requirement that the value returned is the actual version in
+    // the network config will always be called after we've set the last closed
+    // ledger state.
+    if (!mLastClosedLedgerState)
```

Comment:
Could we avoid doing this? This is pretty error-prone, as the callers will have no idea they might be dealing with an incorrect value. Instead, this function should enforce that mLastClosedLedgerState is set, and let the callers deal with scenarios where for whatever reason it's not (or crash to highlight a bug). In case of Maintainer, we start it after we load the LCL, so I think it should be fine.